src/config.rs

```diff
@@ -14,34 +14,48 @@ use crate::{NodeInfo, R};
 use std::collections::HashSet;
 use std::net::IpAddr;
 use std::path::PathBuf;
+use std::str::FromStr;
 use std::{fmt, fs, io};
 
 /// QuicP2p configurations
-#[derive(Debug, Default, Serialize, Deserialize, Eq, PartialEq)]
+#[derive(Debug, Default, Serialize, Deserialize, Eq, PartialEq, StructOpt)]
 pub struct Config {
     /// Hard Coded contacts
+    #[structopt(
+        short,
+        long,
+        default_value = "[]",
+        parse(try_from_str = "serde_json::from_str")
+    )]
     pub hard_coded_contacts: HashSet<NodeInfo>,
     /// Port we want to reserve for QUIC. If none supplied we'll use the OS given random port.
+    #[structopt(short, long)]
     pub port: Option<u16>,
     /// IP address for the listener. If none supplied we'll use the default address (0.0.0.0).
+    #[structopt(long)]
     pub ip: Option<IpAddr>,
     /// This is the maximum message size we'll allow the peer to send to us. Any bigger message and
     /// we'll error out probably shutting down the connection to the peer. If none supplied we'll
     /// default to the documented constant.
+    #[structopt(long)]
     pub max_msg_size_allowed: Option<u32>,
     /// If we hear nothing from the peer in the given interval we declare it offline to us. If none
     /// supplied we'll default to the documented constant.
     ///
     /// The interval is in milliseconds. A value of 0 disables this feature.
+    #[structopt(long)]
     pub idle_timeout_msec: Option<u64>,
     /// Interval to send keep-alives if we are idling so that the peer does not disconnect from us
     /// declaring us offline. If none is supplied we'll default to the documented constant.
     ///
     /// The interval is in milliseconds. A value of 0 disables this feature.
+    #[structopt(long)]
     pub keep_alive_interval_msec: Option<u32>,
     /// Path to our TLS Certificate. This file must contain `SerialisableCertificate` as content
```

Comment:
Ah this is not correct - can we pls change it to say it's not a path we are asking but the certificate itself ?